src/functor.jl

```diff
@@ -192,11 +192,12 @@ julia> fcollect(m, exclude = v -> Functors.isleaf(v))
  Bar([1, 2, 3])
 ```
 """
-function fcollect(x; cache = [], exclude = v -> false)
-  x in cache && return cache
-  if !exclude(x)
-    push!(cache, x)
-    foreach(y -> fcollect(y; cache = cache, exclude = exclude), children(x))
-  end
-  return cache
+function fcollect(x; output = [], cache = Base.IdSet(), exclude = v -> false)
+    x in cache && return output
```

Comment:
I realized it wasn't totally obvious why the code was done this way and also that LegolasFlux will depend on the order (for the same model) not changing in non-breaking releases of Functors.jl, so maybe this comment will help communicate that:
```suggestion
    # note: we don't have an `OrderedIdSet`, so we use an `IdSet` for the cache
    # (to ensure we get exactly 1 copy of each distinct array), and a usual `Vector`
    # for the results, to preserve traversal order (important downstream!).
    x in cache && return output
```
Note the tests already test the order, so hopefully that will be another sign to not break it 😄.